src/jtag3.c

```diff
@@ -1597,6 +1600,27 @@ static int jtag3_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
       }
     }
 
+    else if (str_starts(extended_param, "mode=") &&
+      (str_starts(pgmid, "pickit4") || str_starts(pgmid, "snap"))) {
+      char mode[3];
+      int sscanf_success = sscanf(extended_param, "mode=%3s", mode);
+      if (sscanf_success < 1 || (!str_caseeq(mode, "avr") && !str_caseeq(mode, "pic"))) {
+        pmsg_error("invalid mode setting '%s'\n", extended_param);
```

Comment:
... but `-xmode=avr_rocks` would pass ...

All in all probably better and shorter to `str_caseeq(extended_param, "mode=avr")` for one mode (no sscanf needed) and `str_caseeq(extended_param, "mode=pic")` for the other mode and complain otherwise.

Reply:
Great idea!